### PR TITLE
10.0 website_signup_legal_page_required

### PR DESCRIPTION
**Pull request by eLBati**
based on https://github.com/OCA/website/pull/318

### CODE REVIEW

**Comment by rafaelbn** [comment]
@eLBati this module is already migrate here: https://github.com/OCA/e-commerce/pull/205

**Comment by rafaelbn** [comment]
Sorry mistake ! :smile:

**Comment by pedrobaeza** [comment]
Closing then.

**Comment by pedrobaeza** [comment]
@rafaelbn please avoid the noise.

File: website_signup_legal_page_required/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (SimoRubi) [inline website_signup_legal_page_required/__init__.py:1]: copyright of all __init__file should be as in https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: website_signup_legal_page_required/controllers/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+
```
Comment (SimoRubi) [inline website_signup_legal_page_required/controllers/exceptions.py]: add your own copyright for ordinary Python files as in https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py

File: website_signup_legal_page_required/views/pages.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (SimoRubi) [inline website_signup_legal_page_required/views/pages.xml]: add copyright xml files too as in https://github.com/OCA/maintainer-tools/blob/master/template/module/views/model_name_view.xml

File: website_signup_legal_page_required/views/pages.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+<template id="signup_view"
+          name="Force legal terms"
+          inherit_id="auth_signup.fields"
+          active="True"
```
Comment (SimoRubi) [inline website_signup_legal_page_required/views/pages.xml]: Why is this necessary?

**Review by SimoRubi — APPROVED** [review]
Tried in runbot, code review.
Basically just missing copyrights.

**Review by simahawk — CHANGES_REQUESTED** [review]
@eLBati mostly same remarks as per v9

**Review by chienandalu — COMMENTED** [review]
It would be interesting to register metadata in order to make the acceptance more GDPR compliant as we're doing here: https://github.com/OCA/e-commerce/pull/205/commits/9e6007b14f26076310bf08dd0cc83c76d3d59ba6

**Comment by eLBati** [comment]
@SimoRubi @simahawk @chienandalu thanks, done

File: website_signup_legal_page_required/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            environ = request.httprequest.headers.environ
+            metadata = "Website legal terms acceptance metadata: "
+            metadata += "\n".join(
+                "%s: %s" % (val, environ.get(val)) for val in (
+                    "REMOTE_ADDR",
+                    "HTTP_USER_AGENT",
+                    "HTTP_ACCEPT_LANGUAGE",
+                )
+            )
+            request.env.user.partner_id.message_post(body=metadata)
```
Comment (simahawk) [inline website_signup_legal_page_required/controllers/main.py:28]: just asking: can you do this w/ out sudo on signup?

Comment (chienandalu) [reply website_signup_legal_page_required/controllers/main.py:28]: Seems you can:
image

**Comment by chienandalu** [comment]
LINT is failing somehow :woman_shrugging:

**Comment by pedrobaeza** [comment]
A new check for RST syntax has been enabled as mandatory, so there are README files incorrect in the repository (website_legal_page), and thus the error.

**Comment by eLBati** [comment]
Fixing README with https://github.com/OCA/website/pull/469

**Comment by pedrobaeza** [comment]
Please rebase now that #469 is merged

**Comment by eLBati** [comment]
:green_heart:

File: website_signup_legal_page_required/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+class ForceLegalTerms(AuthSignupHome):
+    def _signup_with_values(self, token, values, *args, **kwargs):
+        """Force accepting legal terms to open an account."""
+        if request.params.get("accepted_legal_terms"):
+            values["accepted_legal_terms"] = True
+            res = super(ForceLegalTerms, self)._signup_with_values(
+                token, values, *args, **kwargs)
+            environ = request.httprequest.headers.environ
+            metadata = "Website legal terms acceptance metadata: "
+            metadata += "\n".join(
```
Comment (yajo) [inline website_signup_legal_page_required/controllers/main.py]: Use `<br/>` instead.

**Review by yajo — CHANGES_REQUESTED** [review]

File: website_signup_legal_page_required/controllers/main.py (hunk 6 above)
Comment (eLBati) [reply website_signup_legal_page_required/controllers/main.py]: @Yajo done

**Review by yajo — APPROVED** [review]

**Comment by eLBati** [comment]
@simahawk what do you say?

**Review by simahawk — APPROVED** [review]
